### PR TITLE
Allow MDS blob parsing to handle invalid entries

### PR DESCRIPTION
**Pull request by sumo**
Allow MDS blob parsing to handle invalid entries. Returns either a certificate error or a These with a list of errors, a registry of metadata entries or both where the MDS has bad entries.

### CODE REVIEW

File: src/Crypto/WebAuthn/Metadata.hs [hunk 1]
```diff
@@ -24,11 +26,10 @@ metadataBlobToRegistry ::
   BS.ByteString ->
   -- | The time at which it was fetched
   HG.DateTime ->
-  -- | Either an error on a registry of metadata entries
-  Either Text Service.MetadataServiceRegistry
+  -- | Either a certifcate error or a list of errors, a registry of metadata entries or both where the MDS has bad entries
+  Either Text (These [Text] Service.MetadataServiceRegistry)
```
Comment (infinisil) [inline src/Crypto/WebAuthn/Metadata.hs:30]:
```suggestion
  Either Text (These (NonEmpty Text) Service.MetadataServiceRegistry)
```

Same for the other cases of `These`

File: webauthn.cabal [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cabal-version: 2.4
 name: webauthn
-version: 0.5.0.1
+version: 0.5.0.2
```
Comment (infinisil) [inline webauthn.cabal:3]: The version should increase to 0.6.0.0 since this is a breaking change to the API.

File: webauthn.cabal [hunk 3]
```diff
@@ -96,6 +96,7 @@ library
     serialise             >= 0.2.3 && < 0.3,
     singletons            >= 2.6 && < 3.2,
     text                  >= 1.2.4 && < 2.1,
+    these                 >= 1 && < 2,
```
Comment (infinisil) [inline webauthn.cabal:99]:
```suggestion
    these                 >= 1.1 && < 1.2,
```

**Review by infinisil — CHANGES_REQUESTED** [review]
Thanks a lot! Please also add an entry to the changelog.md, and rename the commit and this PR to describe the change, not the version.

The demo server will also need to be updated, CI will check for that.

File: webauthn.cabal (hunk 3 above)
Comment (sumo) [reply webauthn.cabal:99]: I've kept upper bounds to 1.3 to align with the other packages that cover multiple minor versions.

**Comment by sumo** [comment]
Changes done!

**Comment by ErinvanderVeen** [comment]
Thank you for resolving all the issues. I'll get to creating a new version as soon as possible.

**Comment by infinisil** [comment]
@sumo Please rename or squash the commits so that there's no commit titled "Version 0.5.0.2"

**Review by ErinvanderVeen — APPROVED** [review]

**Pull request merged**